Add Death indicator fix

DannysPVM opened the pull request:
Fix Issue #3676 where the death indicator plugin was clearing the Bounty Hunter hint arrow. This fix is implemented by changing the Death hint arrow to instead be a tile overlay that marks the players death location as recommended by Adam.

Adam- commented:
I think that a config option for this is a little overkill. The arrow can conflict with content so I would just do the tile overlay.

DannysPVM commented:
I agree @Adam- , I went ahead and fixed this PR so that the plugin will now just show a tile overlay where the player died, instead of complicating things with a hint arrow that can conflict with other content. Let me know if you'd like to see any other changes! 👍

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java:
```diff
@@ -78,12 +78,17 @@
 	private Instant lastDeathTime;
 	private int lastDeathWorld;
 
+	private boolean highlightTile = false;
```
This makes no sense. Just check isDead/the config option.

DannysPVM replied:
Good call, I went ahead and fixed this so that the tile is only rendered if the player hasDied, config.showDeathTileHighlight, and onDeathWorld. I also removed all instance of that highlightTile boolean since it was pointless.

To further clarify:
```java
// Check that the player wants tile highlight, has died, and is on the death world
if (!config.showDeathTileHighlight()
	|| !plugin.hasDied()
	|| !plugin.onDeathWorld())
{
	return null;
}

// Render tile highlight if player wants it, has died, and is on the correct world.
```

 Let me know if there is anything else you'd like changed!

DannysPVM commented:
I think I am done making changes to this PR now. I went ahead and did the following: Fixed the hint arrow conflicting with Bounty Hunter arrow by changing the hint arrow to instead be a tile highlight (as Adam requested), updated the plugin to now use the OverlayManager, fixed issue #3743 by checking if the current time is after the saved death time plus 3600 seconds (1 hour), fixed an unreported issue I found where the markers were showing for any account that I played on locally (not just the account that died) by saving the player who died's name and verifying on login that the current player is the player that is recorded as dead, and added yellow text to the tile highlight overlay and minimap that says "Grave" (like konduit does, iirc). I extensively retested all previous scenario's plus the new ones after the additions and could not find any other issues.

_Current Death Markers_
![image](https://user-images.githubusercontent.com/39487676/41445888-1acde854-7012-11e8-85a1-550de8232ee4.png)

Just let me know if anything needs to be changed. Thanks.

deathbeam commented:
I would still keep hint arrow, but make it disabled by default, but keep it for people that want it even if it interferes with existing stuff.

DannysPVM commented:
This PR is really old, outdated, and all around pretty bad looking back at what I'd done here. I'm just going to close it. I'll consider revisiting the hint arrow as Deathbeam mentioned above, I like that idea.
